apps/omg/lib/omg/state/core.ex

```diff
@@ -18,7 +18,7 @@ defmodule OMG.State.Core do
   All spend transactions, deposits and exits should sync on this for validity of moving funds.
   """
 
-  defstruct [:height, :utxos, pending_txs: [], tx_index: 0, utxo_db_updates: []]
+  defstruct [:height, utxos: %{}, pending_txs: [], tx_index: 0, utxo_db_updates: [], recently_spent: []]
```

Comment:
Let's
1/ revisit the docs to see if there aren't any obsolete stuffs there. This includes `moduledoc` here `docs` here `architecture.md`
2/ let's also expand the `moduledoc` here. I think now it requires a section explaning how the state is managed - what is in db, what in mem, what the semantics are there. Also - what happens if this crashes and restarts.

This part, let's tackle last, after the flows are nailed down

Reply:
Extended @moduledoc - please see if it makes sense to you @pdobacz